junit-jupiter-engine/src/test/java/org/junit/jupiter/engine/extension/TimeoutExtensionTests.java

```diff
@@ -88,6 +89,33 @@ void doesNotApplyTimeoutOnAnnotatedTestMethodsUsingDisabledTimeoutMode() {
 				.isEmpty();
 	}
 
+	@Test
+	@DisplayName("is not applied on annotated @Test methods using timeout mode: disabled")
+	void applyTimeoutOnAnnotatedTestMethodsUsingDisabledOnDebugTimeoutMode() {
+		EngineExecutionResults results = executeTests(request() //
+				.selectors(selectMethod(TimeoutAnnotatedTestMethodTestCase.class, "testMethod")) //
+				.configurationParameter(DEFAULT_TEST_METHOD_TIMEOUT_PROPERTY_NAME, "42ns") //
+				.configurationParameter(TIMEOUT_MODE_PROPERTY_NAME, "disabled_on_debug").build());
+
+		Execution execution = findExecution(results.testEvents(), "testMethod()");
+
+		assertThat(execution.getDuration()) //
+				.isGreaterThanOrEqualTo(Duration.ofMillis(10)) //
+				// The check to see if debugging is pushing the timer just above 1 second
+				.isLessThan(Duration.ofSeconds(2));
+
+		// Should we test if we're debugging? This test will fail if we are debugging.
+		if (RuntimeUtils.isDebug()) {
+			assertThat(execution.getTerminationInfo().getExecutionResult().getThrowable()) //
+					.isEmpty();
+		}
+		else {
+			assertThat(execution.getTerminationInfo().getExecutionResult().getThrowable().orElseThrow()) //
+					.isInstanceOf(TimeoutException.class) //
+					.hasMessage("testMethod() timed out after 10 milliseconds");
+		}
```

Comment:
Comment isn't quite clear. Right now we need to check if we're debugging as the output of the test differs if so. If this is sufficient I'll remove the comment as part of another commit.

Reply:
We'll polish the test a bit after merging it to `master` -- so the comment helps as-is.